### PR TITLE
How to get default system converters

### LINKED ISSUE #23494: Show how to call built-in converters from a custom converter
https://github.com/dotnet/runtime/issues/50205#issuecomment-807137317

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 5de37d79-751f-e372-e866-efde28f4908c
* Version Independent ID: 678c4f23-764a-7ccd-0d6d-159a30957477
* Content: [How to write custom converters for JSON serialization - .NET](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-converters-how-to?pivots=dotnet-5-0#error-handling)
* Content Source: [docs/standard/serialization/system-text-json-converters-how-to.md](https://github.com/dotnet/docs/blob/main/docs/standard/serialization/system-text-json-converters-how-to.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @tdykstra
* Microsoft Alias: **tdykstra**

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #23494
[Internal review - converters doc](https://review.docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-converters-how-to?branch=pr-en-us-29441&pivots=dotnet-7-0#use-default-system-converter)
[Internal review - options doc](https://review.docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-configure-options?branch=pr-en-us-29441&pivots=dotnet-7-0#the-jsonserializeroptionsdefault-property)

### CODE REVIEW

File: docs/standard/serialization/system-text-json-converters-how-to.md [hunk 1]
```diff
@@ -406,6 +429,16 @@ By default, the built-in <xref:System.Text.Json.Serialization.JsonStringEnumConv
 
 ::: zone-end
 
+::: zone pivot="dotnet-7-0"
+
+### Use default system converter
+
+In some scenarios, you might want to use the default system converter in a custom converter. To do that, get the system converter from the <xref:System.Text.Json.JsonSerializerOptions.Default?displayProperty=nameWithType> property, as shown in the following example:
+
+:::code language="csharp" source="snippets/system-text-json-converters-how-to/csharp/GetDefaultConverter.cs" id="Converter" highlight=3-4,17:::
```
Comment (gewarren) [inline docs/standard/serialization/system-text-json-converters-how-to.md:438]:
```suggestion
:::code language="csharp" source="snippets/system-text-json-converters-how-to/csharp/GetDefaultConverter.cs" id="Converter" highlight="3-4,17":::
```

Comment (gewarren) [reply docs/standard/serialization/system-text-json-converters-how-to.md:438]: I'm guessing...

Comment (tdykstra) [reply docs/standard/serialization/system-text-json-converters-how-to.md:438]: Good guess, thanks.

File: docs/standard/serialization/snippets/system-text-json-converters-how-to/csharp/GetDefaultConverter.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+    {
+        return s_defaultConverter.Read(ref reader, typeToConvert, options);
+    }
+}
+// </Converter>
+
+public class Program
+{
+    public static void Main()
```
Comment (eiriktsarpalis) [inline docs/standard/serialization/snippets/system-text-json-converters-how-to/csharp/GetDefaultConverter.cs:37]: Nit: would it make sense to render new code samples using top-level statements?

**Review by eiriktsarpalis — APPROVED** [review]
Thanks!

File: docs/standard/serialization/snippets/system-text-json-converters-how-to/csharp/GetDefaultConverter.cs (hunk 2 above)
Comment (tdykstra) [reply docs/standard/serialization/snippets/system-text-json-converters-how-to/csharp/GetDefaultConverter.cs:37]: Yes, but this part of the code doesn't get rendered in the doc, only the actual converter code that is between the <Converter> tags is shown. This part of the code just makes it possible for the doc writer to test run the snippet. Including Program/Main lets me run a number of snippets in the same project, which I couldn't do with top-level statements. See, for example, https://github.com/dotnet/docs/blob/main/docs/standard/serialization/snippets/system-text-json-how-to-6-0/csharp/Program.cs

**Pull request merged**